_static/link_gen/link.js

```diff
@@ -141,28 +154,58 @@ function changeTab(div) {
  * See https://github.com/git/git/blob/1c52ecf4ba0f4f7af72775695fee653f50737c71/builtin/clone.c#L276
  */
 function generateCloneDirectoryName(gitCloneUrl) {
+    if(gitCloneUrl.slice(-1) == "/")
+        gitCloneUrl = gitCloneUrl.slice(0,-1);
     var lastPart = gitCloneUrl.split('/').slice(-1)[0];
     return lastPart.split(':').slice(-1)[0].replace(/(\.git|\.bundle)?/, '');
 }
 
+function handleSource(args){
+    source = args["source"];
+    branch = "";
+    compressed = true;
+    sourceUrl ="";
+    if(source == "git"){
+        sourceUrl = args["contentRepoUrl"];
+        branch = args["contentRepoBranch"];
+        compressed = false;
+    } else if(source == "googledrive"){
+        sourceUrl = args["driveUrl"];
+    } else if(source == "dropbox"){
+        sourceUrl =  args["dropUrl"];
+    } else if(source == "generic_web"){
+        sourceUrl =  args["webUrl"];
+    }
```

Comment:
Seeing this, I'm thinking "aha, source is the choice of content provider". Perhaps naming this content provider type or similar would be a more suitable name. Not sure about the full picture here, so maybe source makes sense.

Reply:
yes the original piece referred to "git" as the source. I changed everything to be content_provider(or some derivative) and got rid of "source" in both the javascript and HTML attributes